tests/monitoring/metrics.go

```diff
@@ -144,6 +125,7 @@ func fetchPrometheusMetrics(virtClient kubecli.KubevirtClient) *libmonitoring.Qu
 func setupVM(virtClient kubecli.KubevirtClient) {
 	vm := createRunningVM(virtClient)
 	libmonitoring.WaitForMetricValue(virtClient, "kubevirt_number_of_vms", 1)
+	libmonitoring.WaitForMetricValueWithLabelsToBe(virtClient, "kubevirt_vmi_filesystem_capacity_bytes", map[string]string{"namespace": vm.Namespace, "name": vm.Name}, 0, ">", 0)
```

Comment:
This function is not self-explainable and this line makes it worst. It is not clear why we wait for specific metric values and why we delete the VM in this function. 
Can you please separate the test to 2 parts, one for metrics that are based on existing VM, and another for metrics that are based on a VM deletion? Then, I think we can just wait for the metrics existence in the test itself.

Reply:
we would make the test take almost twice as long, maybe we can just refactor the function name and maybe extract some logic to an helper function

Reply:
Why would that make the test so long?

Reply:
Now you propose to:
create -> check some metrics -> delete -> check one metric -> check all the metrics
I propose:
create -> check all metrics expect of deletion based ones -> delete -> check deletion based metrics

Can it work? This way we only check the metrics in the test itself and separate the VMs work from the metrics.

Reply:
ok, let's try to refactor the function, to make it clearer.

Reply:
updated @assafad, let me know what you think